src/shared/sol-conffile.c

```diff
@@ -146,59 +268,60 @@ sol_conffile_load_keyfile_from_dirs(const char *file, char **full_path)
         NULL,
     };
 
-    cwd = getcwd(cwdbuf, sizeof(cwdbuf));
-    if (cwd)
-        search_dirs[0] = cwd;
-    else
-        SOL_DBG("could not getcwd: %d", errno);
-
-    keyfile = g_key_file_new();
-    SOL_NULL_CHECK(keyfile, NULL);
-
     /* absolute path */
-    if (g_path_is_absolute(file)) {
-        if (!g_key_file_load_from_file(keyfile,
-            file, G_KEY_FILE_NONE, &error))
-            goto err;
-    } else if (!g_key_file_load_from_dirs(keyfile, file, search_dirs,
-        full_path, G_KEY_FILE_NONE,
-        &error))
-        goto err;
+    if (file[0] == '/') {
+        *file_reader = sol_file_reader_open(file);
+        if (*file_reader)
+            config_file_contents = sol_file_reader_get_all(*file_reader);
+    } else {
+        for (i = 0; i < ARRAY_SIZE(search_dirs); i++) {
+            char *filename;
+            if (asprintf(&filename, "%s/%s", search_dirs[i], file) <= 0) {
+                SOL_WRN("Couldn't allocate memory for config file.");
+                break;
+            }
+            *file_reader = sol_file_reader_open(file);
+            if (*file_reader)
+                config_file_contents = sol_file_reader_get_all(*file_reader);
 
-    return keyfile;
+            if (config_file_contents.len != 0)
+                break;
+            else
+                SOL_WRN("Error creating type.");
```

Comment:
"Could not open file ($PATH) while creating type" or something like that will help the user when things don't work as expected.